llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -3944,6 +3948,14 @@ LoopVectorizationCostModel::computeMaxVF(ElementCount UserVF, unsigned UserIC) {
   return FixedScalableVFPair::getNone();
 }
 
+bool LoopVectorizationCostModel::useMaxBandwidth(
+    TargetTransformInfo::RegisterKind RegKind) {
+  return MaximizeBandwidth || (MaximizeBandwidth.getNumOccurrences() == 0 &&
+                               (TTI.shouldMaximizeVectorBandwidth(RegKind) ||
+                                (UseWiderVFIfCallVariantsPresent &&
+                                 Legal->hasVectorCallVariants())));
```

Comment:
@preames @wangpc-pp @topperc Making a note here that after this patch we'll likely regress cases like this unless we opt into shouldMaximizeVectorBandwidth https://godbolt.org/z/WcbWGooa4